python/tvm/relay/build_module.py

```diff
@@ -115,6 +117,9 @@ def build(self, mod, target=None, target_host=None, params=None, executor="graph
             - If "graph" is specified, then the graph_executor will be used
             - If "aot" is specified, then the aot_executor will be used
 
+        mod_name: Optional[str]
+            The module name we will build
```

Comment:
I didn't think about any restrictions. The idea was to use the name provided by the user as it is. You are asking something like : `verify("&%!") == false`? If yes, two points:
1. I would only validate. If the name is not valid, an error should be thrown
2. Any suggestions on what makes a "valid" name?

Reply:
cool, i agree with that. mangling is annoying because you have to be 1:1 and doing this may confuse the user. filesystems can handle a lot of strange filenames now--I think C function names are the more restrictive use case of `mod_name` now. Looks like the spec is digits, underscores, letters, and escaped unicode chars. I'd suggest allowing all of that except the escaped unicode. we don't have the restriction on the first char being a digit since we will prefix this.